internal/e2e/resources.go

```diff
@@ -0,0 +1,18 @@
+package e2e
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func GetNonZeroResourceCapacityOrError(node v1.Node, resourceName string) (int, error) {
```

Comment:
naming is redundant when the types imply the behavior. `OrDie` works if theres a `panic` but that's hidden unless you read the implementation
```suggestion
func GetNonZeroResourceCapacity(node v1.Node, resourceName string) (int, error) {
```